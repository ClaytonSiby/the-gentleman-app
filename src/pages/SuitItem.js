import React, { useEffect } from 'react';
import { useSelector, connect } from 'react-redux';
import PropTypes from 'prop-types';
import { useParams } from 'react-router-dom';
import { Container, Col, Row } from 'react-bootstrap';
import getsuitItem from '../redux/suitItem/suitItemActions';
import postAppointment from '../redux/appointments/post/add/addAppActions';
import styles from '../assets/scss/suitItem.module.scss';

const SuitItem = ({ createAppointment, suitItemRequest }) => {
  const authToken = localStorage.getItem('userToken');
  const { id } = useParams();

  useEffect(() => {
    (async () => {
      await suitItemRequest(id, authToken);
    })();
  }, []);

  const suitItemData = useSelector((state) => state.suitItem.suit);

  return (
    <Container className={`${styles.suitContainer}`}>
      <Row>
        <Col
          sm={12}
          md={6}
          className={`${styles.suitImage}`}
          style={{
            background: `linear-gradient(rgba(0,0,0,.4), rgba(0,0,0,.6)), url(${suitItemData.imageUrl})`,
          }}
        />
        <Col sm={12} md={6} className={`${styles.suitDescription} p-3`}>
          <h4>
            {' '}
            {suitItemData.name}
            {' '}
          </h4>
          <div>
            <p>
              available color:&nbsp;
              <span className={`${styles.item}`}>{suitItemData.color}</span>
            </p>
            <p>
              price:&nbsp;$
              <span className={`${styles.item}`}>{suitItemData.price}</span>
            </p>
            <p>
              available size:&nbsp;
              <span className={`${styles.item}`}>{suitItemData.size}</span>
            </p>
            <p>
              type:&nbsp;
              <span className={`${styles.item}`}>{suitItemData.suit_type}</span>
            </p>
            <p>
              <span className={`${styles.item}`}>
                {suitItemData.description}
              </span>
            </p>
            <button
              type="button"
              onClick={() => createAppointment(suitItemData.id, authToken)}
              className="btn btn-md btn-default border"
            >
              Try it Out
            </button>
          </div>
        </Col>
      </Row>
    </Container>
  );
};

SuitItem.propTypes = {
  createAppointment: PropTypes.func.isRequired,
  suitItemRequest: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  createAppointment: (suitId, authToken) => {
    dispatch(postAppointment(suitId, authToken));
  },
  suitItemRequest: (id, token) => dispatch(getsuitItem(id, token)),
});

export default connect(null, mapDispatchToProps)(SuitItem);
