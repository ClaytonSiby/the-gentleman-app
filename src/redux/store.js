import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { persistStore } from 'redux-persist';
import persistReducer from './rootReducer';

const store = createStore(
  persistReducer,
  composeWithDevTools(applyMiddleware(thunk)),
);

const persistor = persistStore(store);

export { store, persistor };
