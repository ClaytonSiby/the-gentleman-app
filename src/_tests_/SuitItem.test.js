import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import SuitItem from '../pages/SuitItem';
import store from './Suits.test';

describe('SuitItem', () => {
  it('Renders the SuitItem page successfully', () => {
    render(
      <Provider store={store}>
        <SuitItem />
      </Provider>,
    );
  });

  test('renders component consistently to the DOM', () => {
    const test = renderer.create(
      <Provider store={store}>
        <SuitItem />
      </Provider>,
    ).toJSON();

    expect(test).toMatchSnapshot();
  });
});
