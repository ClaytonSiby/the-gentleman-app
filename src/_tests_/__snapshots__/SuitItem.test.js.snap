// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuitItem renders component consistently to the DOM 1`] = `
<div
  className="suitContainer container"
>
  <div
    className="row"
  >
    <div
      className="suitImage col-md-6 col-sm-12"
      style={
        Object {
          "background": "linear-gradient(rgba(0,0,0,.7), rgba(0,0,0,.7)), url(undefined)",
        }
      }
    />
    <div
      className="suitDescription p-3 col-md-6 col-sm-12"
    >
      <h4>
         
         
      </h4>
      <div>
        <p>
          available color: 
          <span
            className="item"
          />
        </p>
        <p>
          price: $
          <span
            className="item"
          />
        </p>
        <p>
          available size: 
          <span
            className="item"
          />
        </p>
        <p>
          type: 
          <span
            className="item"
          />
        </p>
        <p>
          <span
            className="item"
          />
        </p>
        <button
          className="btn btn-md btn-default border"
          onClick={[Function]}
          type="button"
        >
          Try it Out
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Suits renders consistently to the DOM 1`] = `
<div
  className="text-light"
>
  <div
    className="loading"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <path transform=\\"translate(2)\\" d=\\"M0 12 V20 H4 V12z\\"> 
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\"  />
  </path>
  <path transform=\\"translate(8)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.2\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\"  />
  </path>
  <path transform=\\"translate(14)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.4\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
  <path transform=\\"translate(20)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.6\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
  <path transform=\\"translate(26)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.8\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
</svg>
",
        }
      }
      style={
        Object {
          "fill": "grey",
          "height": 70,
          "width": 80,
        }
      }
    />
  </div>
</div>
`;
